### ISSUE TITLE
stuck loading into maps with day 2 / loading a new heist after finishing one

### ISSUE THREAD

**Issue by Rideout1** [segabl/pd2-player-randomizer#5]
It seems when ever I have this mod toggled, and I load into a heists day 2, or another heist after completing one, after everyone readys up, the intro for the heist will play, but it wont leave the black screen. Waiting for it to end, or skipping the intro entirely doesn't make a difference. Going into mod options, unticking and reticking the option you have on seems to fix it.

**Comment by Rideout1** [comment]
I tested it a bit more, it also prevents players from syncing into the game. They won't load in unless I retick everything I have set, once I do that they'll all start instantly loading in the background

**Comment by segabl** [comment]
I'll have to check that. by reticking you mean disabling all options and enabling them again?

**Comment by Rideout1** [comment]
Yes. I do this within the loadout screen. Just a simple retick. Its fine for the first heist, but any follow  up day or heist after that makes it so players will not sync until (including yourself it seems) until you retick all the options you have selected. 

Try messing with the melee option if the primary doesn't cause issues for you.

**Comment by segabl** [comment]
Does this only happen with other players in the lobby or also solo?

**Comment by Rideout1** [comment]
Both. If I solo, I can ready up, but I cant get passed the intro. It'll sit on the black screen showing the difficulty and loading spinner, but it wont progress until I press esc, and untick the options and retick them.

As for publics, its a bit different, as any client cant even load in properly (they won t have that % status until I turn it off and turn it back on)

**Comment by segabl** [comment]
Okay, thanks for the info. I'll look into it

**Comment by segabl** [comment]
After trying a few times, I don't seem to be able to reproduce this issue. I have no trouble playing multiple games after one another without being stuck in loading. Maybe some other mod conflicts with this for you?

**Comment by Rideout1** [comment]
Hm, it still seems to happen after removing all my mods. Its just this one that prevents getting into the heist. Though, I guess if its just me with this issue its best to close it. Thanks for taking the time to look into it.

**Comment by segabl** [comment]
What settings do you have the mod at, just out of interest?

**Comment by Rideout1** [comment]
Only weapons from inventory and primary, tbh.

I find that the melee options gives me problems, as its the one I have to untick and retick for it to load after the intro. Even when its originally off.

**Comment by Rideout1** [comment]
I wanted to report back and say I found the cause of the issue, and why I couldn't find it before.

It was the "secondaryasprimary" causing the issue. It seems to soft lock itself when having a secondary in your primary slot. The reason I found this now, and not before, is because the secondary weapons stay in your inventory when you remove the mod, so your mod was probably still trying to call up the secondaries I had. Deleting them from my inventory and the mod itself has fixed the issue.
